frontend/src/wizards/Argo/MultipleSourcesSelector.tsx

```diff
@@ -0,0 +1,227 @@
+import HelmIcon from './logos/HelmIcon.svg'
+import { Channel, getGitBranchList, getGitPathList } from './ArgoWizard'
+import { GitURLPath, HelmURLPath } from './common'
+
+export interface MultipleSourcesSelectorProps {
+  channels: Channel[] | undefined
+  createdChannels: string[]
+  getGitRevisions: (
+    channelPath: string,
+    secretArgs?:
+      | {
+          secretRef?: string
+          namespace?: string
+        }
+      | undefined
+  ) => Promise<unknown>
+  gitChannels: string[]
+  getGitPaths: (
+    channelPath: string,
+    branch: string,
+    secretArgs?:
+      | {
+          secretRef?: string
+          namespace?: string
+        }
+      | undefined
+  ) => Promise<unknown>
+  gitRevisionsAsyncCallback: (() => Promise<string[]>) | undefined
+  helmChannels: string[]
+  setCreatedChannels: Dispatch<SetStateAction<string[]>>
+  gitPathsAsyncCallback: (() => Promise<string[]>) | undefined
+  setGitPathsAsyncCallback: Dispatch<SetStateAction<(() => Promise<string[]>) | undefined>>
+  setGitRevisionsAsyncCallback: Dispatch<SetStateAction<(() => Promise<string[]>) | undefined>>
+  t: TFunction
+}
+
+function combineArraysPromise(promise1: Promise<any[]>, promise2: Promise<any[]>): Promise<any[]> {
+  return Promise.all([promise1, promise2]).then(([result1, result2]) => [...result1, ...result2])
+}
+
+export function MultipleSourcesSelector(props: MultipleSourcesSelectorProps) {
+  const {
+    t,
+    createdChannels,
+    getGitRevisions,
+    gitChannels,
+    getGitPaths,
+    gitRevisionsAsyncCallback,
+    helmChannels,
+    channels,
+    setCreatedChannels,
+    gitPathsAsyncCallback,
+    setGitPathsAsyncCallback,
+    setGitRevisionsAsyncCallback,
+  } = props
+  const editMode = useEditMode()
+  return (
+    <WizArrayInput
+      path="spec.template.spec.sources"
+      placeholder="Add another repository"
+      disallowEmpty={editMode === EditMode.Create}
+      collapsedContent={
+        <Fragment>
+          <WizHidden hidden={(data) => !data.repositoryType}>
+            <Title headingLevel="h6">{t('Type')}</Title>
+          </WizHidden>
+
+          <WizHidden hidden={(data) => data.repositoryType !== 'git'}>
+            <Label style={{ marginRight: 10 }} color="grey">
+              {t('Git')}
+            </Label>
+          </WizHidden>
+          <WizHidden hidden={(data) => data.repositoryType !== 'helm'}>
+            <Label style={{ marginRight: 10 }} color="grey">
+              {t('Helm')}
+            </Label>
+          </WizHidden>
+          <WizTextDetail path="repoURL" placeholder="Expand to enter the repository details" />
+        </Fragment>
+      }
+    >
+      <WizHidden hidden={(data) => data.repositoryType}>
+        <Title headingLevel="h6">{t('Repository type')}</Title>
+      </WizHidden>
+      <WizTiles path="repositoryType">
+        <Tile id="git" value="git" label="Git" icon={<GitAltIcon />} description="Use a Git repository" />
```

Comment:
Does the description need to be translated?

Reply:
Ahh good catch, thanks Feng! I'll add translation here.